### DISCUSSION TITLE
H265 in webview

### DISCUSSION THREAD

**Discussion by vipeweb** [Novage/p2p-media-loader#198] [Q&A]
Hello, I would like to know if there is any way to run h265 on the webview android?

I implemented p2p-media but more than 80% of my audience comes from android, but much of my content is done in h265 in which, as I understand it, Webview does not support it, I would have to implement it in exoplayer or something native, I would have something to do to support the h265?

**Comment by hoangvietfreelancer** [comment]
Could you describe more about it? If it can play on normal browser then it should be play on webview

**Comment by vipeweb** [comment]
I tried to run videos on h265 within a webview but it doesn't seem to work, even though my device is supported

I heard that the Android webview does not support the h265

**Comment by hoangvietfreelancer** [comment]
Can you play it on normal browser?
Also, has you test webview with h264 and what stream type are you using? Dash or HLS?

**Reply by vipeweb** [reply]
tested by webview and also chrome nothing on h265 works, except for native exoplayer app or something like that, mxplayer works perfectly

**Reply by hoangvietfreelancer** [reply]
If your normal browser can't even play it then you will need an h264 version along with h265 to support android users with p2p, otherwise you can only use h265 without p2p on exoplayer or mxplayer or something like that

**Reply by hoangvietfreelancer** [reply]
Or maybe you can try #52

**Comment by vipeweb** [comment]
I'm using it through a WebView, but I can hardly use h265 because the codec is not included in chromium, which is the webview engine, I was satisfied for a while but I'll have to run after some way to integrate with exoplayer, if I find out something I will share here.

**Comment by mrlika** [comment]
HEVC\H.265 should be supported by hardware (SoC). That is why not all Android phones (low-end) support HEVC. Could you consider using H.264 as a fallback?

We are glad to announce that p2p-media-loader@v1 has been released. It now works in iPhone webview and features HEVC support there.

We will publish our Flutter demo, which compiles and works on Android and Apple devices.

Check for reference: https://caniuse.com/hevc

**Reply by vipeweb** [reply]
Hi @mrlika , good to hear you're back, I thought you had given up.

We use native Android apps, Java or Kotlin, is any implementation possible?

**Reply by mrlika** [reply]
Yes. Flutter demo is just for reference. Main components of the demo are WebView and a HTML5 video player with P2P in it.

**Comment by vipeweb** [comment]
Your solution will help iPhone,IOS
but for Android I believe that only a proxy to be able to run natively on exoplayer would be the solution
Em quinta-feira, 27 de junho de 2024 às 01:54, Andriy Lysnevych ***@***.***> escreveu:

**Comment by DimaDemchenko** [comment]
You can try to use [AV1](https://caniuse.com/av1) on most of the devices and fall back to H.265 on Apple devices with chips older than M3/A17. This approach should provide the best video compression.
